lib/openhab/core/rules/rule.rb

```diff
@@ -163,11 +163,22 @@ def to_s
         # Manually trigger the rule
         #
         # @param [Object, nil] event The event to pass to the rule's execution blocks.
-        # @return [void]
-        #
-        def trigger(event = nil)
-          Rules.manager.run_now(uid, false, { "event" => event })
+        # @param [Boolean] consider_conditions Whether to check the conditions of the called rules.
+        # @param [kwargs] context The context to pass to the conditions and the actions of the rule.
+        # @return [Hash] A copy of the rule context, including possible return values.
+        #
+        def trigger(event = nil, consider_conditions: false, **context)
+          begin
+            event ||= org.openhab.core.automation.events.AutomationEventFactory
+                         .createExecutionEvent(uid, nil, "manual")
+          rescue NameError
+            # @deprecated OH3.4 doesn't have AutomationEventFactory
+          end
+          context.transform_keys!(&:to_s)
+          context["event"] = event if event # @deprecated OH3.4 - remove if guard. In OH4 `event` will never be nil
+          Rules.manager.run_now(uid, consider_conditions, context)
```

Comment:
we should unwrap any proxies here, so that raw items, etc. get passed through. otherwise we may end up with double-wrapped items on the other side (or completely useless objects, if the other side isn't ruby, or is a different instance of a ruby engine)

Reply:
good point!